sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/PlacementRule.java

```diff
@@ -46,8 +46,17 @@
      * Must be explicitly implemented by all PlacementRules.
      *
      * @see com.mesosphere.sdk.offer.TaskUtils#areDifferent(
-     * com.mesosphere.sdk.specification.TaskSpec,
+     *com.mesosphere.sdk.specification.TaskSpec,
      * com.mesosphere.sdk.specification.TaskSpec)
      */
     boolean equals(Object o);
+
+    /**
+     * All placement rules are valid by default. This can be overridden as
+     * required.
+     */
+    @JsonIgnore
+    default boolean isValid() {
```

Comment:
Instead of adding a dedicated function for checking validity, how about just having an `instanceof InvalidPlacementRule` check in the config validator? There's already some precedent for this with `instanceof PassthroughRule` in a few places. Could do something like this in the validator:

```
private static boolean isValid(PlacementRule rule) {
  if (rule instanceof OrRule) {
    return ((OrRule) rule).getRules().stream().allMatch(r -> isValid(r));
  } else if (rule instanceof AndRule) {
    return ((AndRule) rule).getRules().stream().allMatch(r -> isValid(r));
  } else {
    return !(rule instanceof InvalidPlacementRule);
  }
}
```

Reply:
Not that this is necessarily the best solution, but I feel wary about modifying the PlacementRule interface for our internal validation purposes.

Reply:
Yes, I wasn't 100% happy with the new method in the interface, but also don't think that the `instanceof` check is ideal (there is only a single "valid" use of it for `PassthroughRule`). I've decided to use your suggestion as this is the only place where we need to then define valid placement rules.